node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java

```diff
@@ -210,13 +194,28 @@ private ConfigServerApiImpl(Collection<URI> configServerHosts, SelfCloseableHttp
         }, wantedReturnType);
     }
 
+    @Override
+    public void close() {
+        runOnClose.run();
+        client.close();
+    }
+
     private void setContentTypeToApplicationJson(HttpRequestBase request) {
         request.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
     }
 
-    @Override
-    public void setSSLConnectionSocketFactory(SSLConnectionSocketFactory sslSocketFactory) {
-        this.client = new SelfCloseableHttpClient(sslSocketFactory);
+    private void setClient(SSLContext sslContext, AthenzService identity) {
+        this.client = createClient(sslContext, identity);
```

Comment:
You are leaking the connection pool of the old client here as you are not explicitly closing it. Probably not a big deal since the ssl context is not updated often though.

Reply:
It cannot be closed as there may be asynchronous (or future) executions on the client's execute(). The pool is closed in finalize().